### PR TITLE
Userstory3

### PR DESCRIPTION
**Pull request by DenisKorkmaz**
Please check :)

### CODE REVIEW

File: components/ThreeDotsMenü/ThreeDotMenü.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+          </StyledLI>
+          <StyledLI>
+            <Link
+              href={`/summary?shareURL=${encodeURIComponent(
+                article.shareURL
+              )}&teaserImage=${encodeURIComponent(
+                article.teaserImage?.imageVariants?.["16x9-1920"] || ""
+              )}`}>
+              <StyledLink>Zusammenfassung</StyledLink>
+            </Link>
```
Comment (Roland-Hufnagel) [inline components/ThreeDotsMenü/ThreeDotMenü.js]: Here you have a ``<a>``tag inside a ``<Link>``tag. What's the reason?

**Review by Roland-Hufnagel — APPROVED** [review]
Please make sure to fit your userstory. I cannot see a real summery in the vercel-preview.

**Review by Roland-Hufnagel — CHANGES_REQUESTED** [review]
Please make sure to fit your userstory. I cannot see a real summery in the vercel-preview.

**Comment by DenisKorkmaz** [comment]
@Roland-Hufnagel the API has a limit of 100 requests. To ensure that no empty data is displayed, I implemented default text as a fallback. Tomorrow, 100 new requests should be possible.

File: pages/styles.css
Comment (Kheirah) [inline pages/styles.css:1]: Instead of raw CSS use `styled-components` for styling.

File: components/ThreeDotsMenü/ThreeDotMenü.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        newFavorites = newFavorites.filter(
+          (favorite) => favorite.shareURL !== article.shareURL
+        );
+      } else {
+        newFavorites.push(article);
+      }
+      setFavorites(newFavorites);
+      setIsFavorite(!isFavorite);
+    }
+  }
```
Comment (Kheirah) [inline components/ThreeDotsMenü/ThreeDotMenü.js]: You don't need the outer braces.
```suggestion
      if (isFavorite) {
        newFavorites = newFavorites.filter(
          (favorite) => favorite.shareURL !== article.shareURL
        );
      } else {
        newFavorites.push(article);
      }
      setFavorites(newFavorites);
      setIsFavorite(!isFavorite);
    }
```

File: components/ThreeDotsMenü/ThreeDotMenü.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  const [favorites, setFavorites] = useLocalStorageState("favorites", []);
+  const [isFavorite, setIsFavorite] = useState(false);
+
+  useEffect(() => {
+    if (!favorites) return;
+    const found = favorites.some(
+      (favorite) => favorite.shareURL === article.shareURL
+    );
+    setIsFavorite(found);
+  }, [favorites, article.shareURL]);
```
Comment (Kheirah) [inline components/ThreeDotsMenü/ThreeDotMenü.js]: The state of marked favorites should not be specific to the `ThreeDotMenu` component.
The component shouldn't care about the list of favorites. It should only be responsible for opening and closing its menu. This is called **separation of concerns.** The component shouldn't be concerned with managing the localStorage state.

Apart from that, there's another reason which is technical: The `ThreeDotsMenu` component is rendered once for every article you display. If you display 20 articles you render this component 20 times. And for every one of these 20 `ThreeDotMenus` you have a separate `state` and `useEffect`. But you only need to manage your state _once._

Lift up the state and the handle functions to a higher level and manage your state from there. Pass down your state to components that rely on it, like this component.

File: components/ThreeDotsMenü/ThreeDotMenü.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+              isFavorite={isFavorite}
+              handleMarkFavorite={handleMarkFavorite}
+            />
+          </StyledLI>
+          <StyledLI>
+            <Link
+              href={article.shareURL}
+              passHref
+              target="_blank"
+              rel="noopener noreferrer">
```
Comment (Kheirah) [inline components/ThreeDotsMenü/ThreeDotMenü.js:25]: Very good! Security comes first.

File: components/ThreeDotsMenü/styles.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
+import styled from "styled-components";
+
+export const StyledLink = styled.a`
```
Comment (Kheirah) [inline components/ThreeDotsMenü/styles.js]: Is there a reason why this is a normal anchor tag instead of the `Link` component?

**Review by Kheirah — CHANGES_REQUESTED** [review]
Well done!

Your app functions properly, but you could improve on how you manage the state. Currently it's managed by a component. Lift up your state and its management to handle state only at one place instead of at the component level which renders at several places.

Also have a look at the css file and switch to using `styled-components`.

**Review by Roland-Hufnagel — APPROVED** [review]
Hi Denis, you lifted up the state one level, but the ArticleCard is also rendered twenty times or more. Since Kheirah mentioned the separation of concerns, you should lift it up one more level. But it is up to you to implement this.

**Pull request merged**